Missing `EditorSceneImporterMesh` and `EditorSceneImporter` when building with Mono

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
36c943260ed40c6e31a6ecff7f6e1674714b4fcb


**OS/device including version:**
Pop!_OS 20.10


**Issue description:**

Building the editor with Mono enabled fails, complaining that `EditorSceneImporterMesh` and `EditorSceneImporter` are missing. The files pertaining those classes seem to have been generated just fine under `modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GodotObjects/`.

<details>
<summary>Build log</summary>

```
scons: Reading SConscript files ...
Enabling ALSA
Enabling PulseAudio
Mono root directory not found. Using pkg-config instead
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
build_api_solution(["bin/GodotSharp/Api/Debug/GodotSharp.dll", "bin/GodotSharp/Api/Debug/GodotSharp.pdb", "bin/GodotSharp/Api/Debug/GodotSharp.xml", "bin/GodotSharp/Api/Debug/GodotSharpEditor.dll", "bin/GodotSharp/Api/Debug/GodotSharpEditor.pdb", "bin/GodotSharp/Api/Debug/GodotSharpEditor.xml"], [])
MSBuild path: /usr/bin/dotnet
Microsoft (R) Build Engine version 16.8.0+126527ff1 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/EditorSceneImporterGLTF.cs(9,52): error CS0246: The type or namespace name 'EditorSceneImporter' could not be found (are you missing a using directive or an assembly reference?) [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/NativeCalls.cs(3033,32): error CS0246: The type or namespace name 'EditorSceneImporterMesh' could not be found (are you missing a using directive or an assembly reference?) [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GLTFAccessor.cs(351,20): warning CS0108: 'GLTFAccessor.GetType()' hides inherited member 'object.GetType()'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GLTFLight.cs(163,23): warning CS0108: 'GLTFLight.GetType()' hides inherited member 'object.GetType()'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GLTFMesh.cs(11,16): error CS0246: The type or namespace name 'EditorSceneImporterMesh' could not be found (are you missing a using directive or an assembly reference?) [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GLTFMesh.cs(59,16): error CS0246: The type or namespace name 'EditorSceneImporterMesh' could not be found (are you missing a using directive or an assembly reference?) [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GLTFMesh.cs(69,29): error CS0246: The type or namespace name 'EditorSceneImporterMesh' could not be found (are you missing a using directive or an assembly reference?) [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/GraphNode.cs(54,38): warning CS0108: 'GraphNode.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/RichTextLabel.cs(340,38): warning CS0108: 'RichTextLabel.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Label.cs(77,38): warning CS0108: 'Label.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/TextEdit.cs(188,38): warning CS0108: 'TextEdit.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/ArrayMesh.cs(89,36): warning CS0108: 'ArrayMesh.BlendShapeMode' hides inherited member 'Mesh.BlendShapeMode'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/LineEdit.cs(428,38): warning CS0108: 'LineEdit.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/LinkButton.cs(53,38): warning CS0108: 'LinkButton.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Button.cs(81,38): warning CS0108: 'Button.TextDirection' hides inherited member 'Control.TextDirection'. Use the new keyword if hiding was intended. [/home/rphln/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
scons: building terminated because of errors.
```
</details>

**Steps to reproduce:**
Following the instructions outlined at the documentation:

```bash
scons p=linuxbsd tools=yes module_mono_enabled=yes mono_glue=no
bin/godot.linuxbsd.tools.64.mono --generate-mono-glue modules/mono/glue
scons p=linuxbsd target=release_debug tools=yes module_mono_enabled=yes
```

The error happens during the third command.

**Minimal reproduction project:**
N/A
<!-- A small Godot project which reproduces the issue. Drag and drop a zip archive to upload it. -->

--- a/modules/gltf/register_types.cpp
+++ b/modules/gltf/register_types.cpp
@@ -60,9 +60,10 @@ static void _editor_init() {
 void register_gltf_types() {
 #ifndef _3D_DISABLED
 #ifdef TOOLS_ENABLED
-	ClassDB::register_class<EditorSceneImporterGLTF>();
 	ClassDB::APIType prev_api = ClassDB::get_current_api();
 	ClassDB::set_current_api(ClassDB::API_EDITOR);
+	ClassDB::register_class<EditorSceneImporterGLTF>();
+	ClassDB::register_class<GLTFMesh>();
 	EditorPlugins::add_by_type<SceneExporterGLTFPlugin>();
 	ClassDB::set_current_api(prev_api);
 	EditorNode::add_init_callback(_editor_init);
@@ -75,7 +76,6 @@ void register_gltf_types() {
 	ClassDB::register_class<GLTFTexture>();
 	ClassDB::register_class<GLTFSkeleton>();
 	ClassDB::register_class<GLTFSkin>();
-	ClassDB::register_class<GLTFMesh>();
 	ClassDB::register_class<GLTFCamera>();
 	ClassDB::register_class<GLTFLight>();
 	ClassDB::register_class<GLTFState>();

